Dockerfile

```diff
@@ -0,0 +1,28 @@
+# Build Hapistrano
+FROM fpco/stack-build as build-env
+
+MAINTAINER Javier Casas <jcasas@stackbuilders.com>
+
+WORKDIR /hapistrano
+COPY stack.yaml .
+COPY hapistrano.cabal .
+RUN stack build --only-dependencies
+
+COPY src/ src/
+COPY app/ app/
+COPY script/ script/
+COPY LICENSE .
+COPY Setup.hs .
+RUN stack install
+
+# Copy Hapistrano to a basic Debian
+FROM debian:stretch-slim
+RUN apt-get update && \
+    apt-get install -y libgmp-dev && \
+    rm -rf /var/lib/apt/lists/*
+
+COPY --from=build-env /root/.local/bin/hap /bin
+
+# To ensure it works on Debian
+RUN hap --version
+CMD /bin/hap
```

Comment:
I think it's better to use [`ENTRYPOINT`](https://docs.docker.com/engine/reference/builder/#entrypoint). Per the documentation:

> An ENTRYPOINT allows you to configure a container that will run as an executable. 

So we could just execute: `docker run -it --rm hap`